### DISCUSSION TITLE
Charging Range (i.e. discharge until 50% and charge until 80%)

### DISCUSSION THREAD

**Discussion by petter1** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#84] [Ideas]
I would love a feature, that would allow me to set a range. For example if I plug in my Mac with 70% that it would charge until it has 80% then run on power an let the Akku fall to 50% over time. when it is at 50% it should charge it back to 80%. This would be my HomeOffice setup.

**Comment by MatthiasKerbl** [comment]
Hello petter1,
why do you want to do this with your battery?
As far as I know, there is no research that indicates a longer battery life if you do these small cycles instead of no cycles at all.
Do you have any data that points to the benefits of this type of charging?

**Reply by petter1** [reply]
Well i thought about charging to 80% and then don’t touch the battery at all. I thought that the battery discharges over time on its own. So I thought if the Charge of the battery  falls to 79% it would start charging again. So I wanted a bigger range so that it resolves in less cycles.

**Reply by MatthiasKerbl** [reply]
Now I get what you meant. Interesting idea, I'll look into it and talk to David about it.

**Reply by petter1** [reply]
👏🏻vielen Dank fürs hinzufügen in die pro Version 😁

**Reply by MatthiasKerbl** [reply]
> 👏🏻vielen Dank fürs hinzufügen in die pro Version 😁

Sehr gerne, und danke für die gute Idee. :)

**Reply by Giel9988** [reply]
Hello,

Isn't this called 'sailing' in the app? (was this conversation held before the addition of sailing? (apologies if I'm mistaken))

Another question, what is the best for a Li-ion battery, leaving a mac plugged at all times and set the charging limit to 80%? 

I think I understood correctly that it will discharge automatically or that a bit of charge might be used as a buffer when a lot of power is needed. This results in tiny 79% - 80% cycles, what is the impact of this on the battery? Is it bad? (again apologies if this was mentioned somewhere here or on the internet, could not find an answer to this)

Are bigger cycles better? Like 50%-80%? 
 
> Well i thought about charging to 80% and then don’t touch the battery at all. I thought that the battery discharges over time on its own. So I thought if the Charge of the battery falls to 79% it would start charging again. So I wanted a bigger range so that it resolves in less cycles.

I have another remark:

The link to the article by Battery University does not work on the AppHouseKitchen site,.
In the article they mention: "Li-ion is maintenance-free and the battery lasts longest when operating between 30 and 80 percent SoC." 

But on the AppHouseKitchen something different is mentioned: "Li-ion and polymer batteries (like the one in your MacBook) last the longest when operated between 20% and 60%. "

- Is there a similar plan for an ios app? Or is there an app already by AppHouseKitchen/ someone else? (for jailbroken iPhone of course) 
- How can I prevent (or have warning pop up when my goes below a curtain %) my mac from going below 20% or 30%? (or whatever the best percentage is)
- 
Sorry for any stupid things in here, I do not know much about batteries, I do know that this app is amazing! 


Thank you very much!

**Reply by Giel9988** [reply]
Sorry! Just found the blog post on Sailing mode. Some of my questions have been answered, not all.

Will buy the Pro Version. So happy with Sailing mode!
Would love to see new developments on this one day: 'Which Interval Should I Choose For My Battery To Last As Long As Possible?'

Kind regards,

Giel

**Comment by DmacMcgreg** [comment]
I'm not quite sure I understand how sailing mode works. After reading the documentation, it should be as easy as turning sailing mode on, and once it reaches the desired charge, it should prevent any further charging until the sailing mode value is reached, and then charging will resume.

So if I set my limit at 70%, then with sailing 10% it should turn off the charging and naturally go down to 60%. Is that correct? Because until now I have not been able to verify this. Am I doing something wrong?

I have a macbook pro 2019 16" with 97W USB C charging.

**Reply by MatthiasKerbl** [reply]
Hi @DmacMcgreg,

have you read our blog post on the Sailing Mode yet?
https://apphousekitchen.com/feature-explanation-sailing-mode/

**Reply by DmacMcgreg** [reply]
@MatthiasKerbl  Thanks, I did read it, but I'm still super confused as to how it will ever reach the lower limit in this case.

**Reply by MatthiasKerbl** [reply]
There are two reasons why your battery will lose charge over time. I described it in the article:
"However, small discharges of the battery still occur when a lot of power is needed for a short time. Then the battery of the MacBook steps in as a buffer. If the charge level drops below the charge limit due to these short discharges, AlDente recharges the battery to the desired charge limit. A second effect on why the battery loses charge is time. If a battery is left for weeks without charging or discharging, it will also lose charge over time."

Hope that helps. :)

**Reply by DmacMcgreg** [reply]
Hi @MatthiasKerbl 
Sorry, I've already read the article a few times and it still doesn't make any sense to me.
If I leave the battery without charging, of course it will discharge, but then how does sailing mode do anything? 
I've never come across a situation where my laptop has ever lost charge while being plugged in, not on any of my macbooks. 

So I'm just not seeing the point of sailing mode and how it can be possible. 
1. If I let the battery discharge because it is unplugged, then sailing mode cant do anything since there is no power available to charge it.
2. If the battery is sitting at the max level, and sailing mode is turned on, then since the workload of the computer will never be more than what the power adapter can give it, then the battery will never drop from its max level set in the AlDente app

I was expecting sailing mode to just outright turn off the charging once it has reached my max charge rate, and continue to drop no matter what until it has reached its lower limit, but that doesn't seem to be the case. Wouldn't that be a much more interesting use case?

Please let me know if I'm missing something here. Apologies if I'm coming across as a bit frustrated, its just I paid for the pro version specifically assuming that this was what it was supposed to do, and I seem to be mistaken here or something.

**Reply by MatthiasKerbl** [reply]
Hi @DmacMcgreg,

> If I leave the battery without charging, of course it will discharge, but then how does sailing mode do anything?

Normally, if your MacBook SoC drops from the set charge limit 80% to 79% AlDente will charge your MacBook again to 80%. However, with Sailing Mode activated, AlDente will wait until the SoC has reached the lower limit, such as 75%. This would theoretically be better than charging 1% over and over again. 

> I've never come across a situation where my laptop has ever lost charge while being plugged in, not on any of my macbooks.

That's because if your MacBook loses any charge your MacBook will normally immediately charge back up to 100%. Without you noticing anything.

> If the battery is sitting at the max level, and sailing mode is turned on, then since the workload of the computer will never be more than what the power adapter can give it, then the battery will never drop from its max level set in the AlDente app

Actually, your MacBook has power spikes which succeed the power of your power adapter. And as mentioned above, the battery will lose some charge over time on its own.

> I was expecting sailing mode to just outright turn off the charging once it has reached my max charge rate, and continue to drop no matter what until it has reached its lower limit, but that doesn't seem to be the case. Wouldn't that be a much more interesting use case?

We could definitely include a feature like this but we intentionally did not. A feature like this would only add cycles to your battery, which would result in a faster battery degradation, and the whole point of AlDente is to slow battery degradation.
Maybe I understand something wrong. Why would you want a feature like this?

It is all good. We are always trying to improve.

**Reply by DmacMcgreg** [reply]
Dear @MatthiasKerbl 
It was my understanding that there will be less heat and pressure on the battery if you allow it to normally discharge to a lower range. Will the battery really last longer by continually keeping it at 70% or so? Or is it better to allow it to discharge to 30% or so. 

The main reason why I was looking for sailing mode was precisely because I was under the assumption that allowing the battery to periodically do this discharging was healthy, since constantly having the power adapter pumping charge into the battery is not healthy, which it seems to be doing when it keeps it at the desired charge amount.

Are my assumptions faulty here? Has anyone done any research to determine which of the two cases is actually better?

I'm still not able to get the battery to ever really go below the set amount by using sailing mode, so I'm still really confused about how it's supposed to work, sorry about that. Maybe I'm not the intended customer for this specific feature, since it seems in order for it to be useful, the computer needs to be doing intensive tasks, which mine doesn't seem to be doing.

**Reply by dmelgar** [reply]
Apple used to have an article describing lithium ion battery health. I don't have the quote, but they generally said that lithium-ion batteries like being used, ie having energy flow in/out. That's why when Apple laptops charge to 100% they start discharging until they reach around 95% and start charging back up. You may not be able to see it with the menu bar battery indicator since it will report 100% even if the battery is really 95%. If you use something like Coconut battery it will clearly show this behavior.

With Aldente, I had hoped to see similar behavior. In my case I have it set to 75%. I had hoped to see it range from 70-75% but it does not. That's what I was assuming sailing mode would allow.

I have not been able to find another lithium-ion reference to state this. The Apple page has been removed or replaced with a less detailed version which no longer describes the behavior between 95-100%.
